feature:  support to enable HTTPS for admin API

dabue opened the pull request:
### Summary

#1647 
support to enable HTTPS for admin API 
1. add https_admin in config.yaml
2. update bin/apisix
3. add crt files
4. update apisix_cli_test.sh
Fix #1647

dabue commented:
> please add test cases for this feature

I will add the test cases.

membphis commented on bin/apisix:
```diff
@@ -286,7 +286,18 @@ http {
 
     {% if enable_admin and port_admin then %}
     server {
+        {%if https_admin then%}
```
I think we need to update the `conf/config.yaml` file, add new field `https_admin`

dabue commented:
I have updated it and added the test cases.



88786912@qq.com
 
From: YuanSheng Wang
Date: 2020-06-05 21:30
To: apache/incubator-apisix
CC: dabue; Author
Subject: Re: [apache/incubator-apisix] [feature] support to enable HTTPS for admin API (#1648)
@membphis commented on this pull request.


In bin/apisix:
> @@ -286,7 +286,18 @@ http {
 
     {% if enable_admin and port_admin then %}
     server {
+        {%if https_admin then%}

I think we need to update the conf/config.yaml file, add new field https_admin
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

membphis commented on conf/config.yaml:
```diff
@@ -54,6 +54,7 @@ apisix:
     - 127.0.0.0/24              # If we don't set any IP list, then any IP access is allowed by default.
   #   - "::/64"
   # port_admin: 9180              # use a separate port
+  # https_admin: true             # enable https
```
need more comments:

```
                                # enable HTTPS when use a separate port for Admin API.
                                # Admin API will use conf/apisix_admin_api.crt and conf/apisix_admin_api.key as certificate.
```

moonming commented on .travis/apisix_cli_test.sh:
```diff
@@ -72,3 +72,47 @@ done
 
 sed -i '/dns_resolver:/,+4s/^#//'  conf/config.yaml
 echo "passed: system nameserver imported"
+
```
I think we should add test cases in `t/admin/`.
like https://github.com/apache/incubator-apisix/blob/master/t/admin/stream-routes-disable.t#L37 to change `config.yaml`, and call https request like https://github.com/apache/incubator-apisix/blob/master/t/plugin/redirect.t#L608

membphis replied:
I think the current implementation is better.

The `conf/nginx.conf` is generated by `conf/config.yaml`, so we use `CLI` to test this feature is better.

The `conf/nginx.conf` in `Test:Nginx` was generated by `t/APISIX.pm`, it is very different for a real user.

membphis replied:
@moonming wait for your response

moonming replied:
I think the test coverage of add test case in shell is too low, e.g. I don't know what happens if I open https but send an http request, and it not test that the host is incorrect, it's basically the same as not testing at all.
Maybe we can merge this PR first, but test cases must added later.

dabue commented:
The test cases only checked whether the added field  https_admin in config.yaml is in effect or not.
